src/config/useMergedTheme.ts

```diff
@@ -0,0 +1,45 @@
+import { createTheme, Theme, ThemeOptions } from '@mui/material';
+// This is what MUI uses internally to merge the themes in createTheme
+import deepmerge from '@mui/utils/deepmerge';
+import { useEffect, useState } from 'react';
+
+import { baseThemeDef, fullTheme, fullThemeDef } from './theme';
+import { fetchTheme } from './utils';
+
+import useLocalStorage from '@/hooks/useLocalStorage';
+
+const useMergedTheme = (remoteThemeId?: string) => {
+  const [remoteThemeOptions, setRemoteThemeOptions] = useLocalStorage<
+    ThemeOptions | undefined
+  >('remoteTheme', undefined);
+  const [mergedTheme, setMergedTheme] = useState<Theme>(fullTheme);
+
+  useEffect(() => {
+    if (!remoteThemeId) return;
+    fetchTheme(remoteThemeId).then((data) =>
+      setRemoteThemeOptions(data as ThemeOptions)
+    );
+  }, [remoteThemeId, setRemoteThemeOptions]);
+
+  useEffect(() => {
+    let newTheme = fullTheme;
+
+    try {
+      const merged = createTheme(
+        deepmerge(
+          deepmerge(baseThemeDef, fullThemeDef),
+          remoteThemeOptions || {}
+        )
+      );
```

Comment:
Ah @ZachCMP sorry I missed this the first time. This is not allowing us to "compose" themes. The reason I split up the "base" and the theme in the way I did is to allow for theme composition: see [Theme composition: using theme options to define other options](https://mui.com/material-ui/customization/theming/#createtheme-options-args-theme).

I was trying to figure out how this was feasible on your branch, and I basically ended up implementing that, so I pushed that up.. sorry for no warning on that! Can you take a look at https://github.com/greenriver/hmis-frontend/pull/163/commits/a02358a554147605398d6b2262668adcb952064f and let me know what you think?

I am looking at the outlined MuiButtons as an example. On the Enrollment page there are several. In that commit, I updated it to set the background for those to `theme.palette.background.paper`. In order for that to work, we need the actual composition. Deepmerging everything won't do the trick. In the "halloween" they successfully have a black background.

Reply:
@gigxz Gotcha. I looked at your changes and they look good to me. The most important part is that the base theme and the remote theme are combined via `deepmerge` and not `createTheme`, which you have implemented there. So that all looks good to me.